### PR TITLE
refactor the structure of the semantics

### PR DESCRIPTION
**Pull request by yzhang90**

### CODE REVIEW

File: elrond.md [hunk 1]
```diff
@@ -1,125 +1,13 @@
...
-      [priority(10)]
-
-endmodule
+require "wasm-text.md"
+require "wasm-coverage.md"
 ```
 
-## Elrond Node
-
+Erlond Node
```
Comment (hjorthjort) [inline elrond.md:9]:
```suggestion
Elrond Node
```

File: elrond.md [hunk 2]
```diff
@@ -209,13 +97,94 @@ The length is the number of bytes the argument represents.
...
+When `AUTO-ALLOCATE` is imported, an new module will be automatically created and registered whenever necessary to resolve an import.
+This makes it possible to implement host modules easily in K.
+Accessing the import will result in an instruction being left on the `instrs` cell that can't be resolved in the regular Wasm semantics.
+Instead, the embedder can add rules for handling the host import.
+
+Currently, only function imports are supported.
+Calling an imported host function will result in `hostCall(MODULE_NAME, FUNCTION_NAME, FUNCTION_TYPE)` being left on the `instrs` cell.
+
+```k
+module AUTO-ALLOCATE
```
Comment (hjorthjort) [inline elrond.md:116]: Maybe it's better if we call this `WASM-AUTO-ALLOCATE`, as a form of namespacing

File: elrond.md [hunk 3]
```diff
@@ -209,13 +97,94 @@ The length is the number of bytes the argument represents.
...
+ // ----------------------------------------------------
+    rule wasmString2StringStripped(WS) => #stripQuotes(#parseWasmString(WS))
+
+    rule #stripQuotes(S) => replaceAll(S, "\"", "")
+
 endmodule
 ```
 
-## Connecting Node and Wasm
+Combine Elrond Node With WASM
```
Comment (hjorthjort) [inline elrond.md:186]:
```suggestion
Combine Elrond Node With Wasm
```

File: elrond.md [hunk 4]
```diff
@@ -248,9 +221,9 @@ module ELROND
          <bytesStack> _ : STACK => STACK </bytesStack>
 ```
 
-### Sync With WASM VM
+### Node And WASM VM Synchronization
```
Comment (hjorthjort) [inline elrond.md:224]:
```suggestion
### Node And Wasm VM Synchronization
```

File: elrond.md [hunk 5]
```diff
@@ -248,9 +221,9 @@ module ELROND
...
 
-`#endWASM` waits for the WASM VM to finish the execution and check the return code.
+- `#endWASM` waits for the WASM VM to finish the execution and check the return code.
```
Comment (hjorthjort) [inline elrond.md:226]:
```suggestion
- `#endWasm` waits for the Wasm VM to finish the execution and check the return code.
```

File: elrond.md [hunk 6]
```diff
@@ -248,9 +221,9 @@ module ELROND
...
 
 ```k
     syntax InternalCmd ::= "#endWASM"
```
Comment (hjorthjort) [inline elrond.md:229]:
```suggestion
    syntax InternalCmd ::= "#endWasm"
```

File: elrond.md [hunk 7]
```diff
@@ -847,7 +819,7 @@ The (incorrect) default implementation of a host call is to just return zero val
                  => pushWorldState
                  ~> transferFunds(FROM, TO, VALUE)
                  ~> mkCall(FROM, TO, VALUE, FUNCNAME, ARGS, GASLIMIT, GASPRICE)
-                 ~> #endWASM                
+                 ~> #endWASM
```
Comment (hjorthjort) [inline elrond.md:822]:
```suggestion
                 ~> #endWasm
```

File: wasm-coverage.md [hunk 8]
```diff
@@ -0,0 +1,37 @@
+WASM Code Coverage
```
Comment (hjorthjort) [inline wasm-coverage.md:1]:
```suggestion
Wasm Code Coverage
```

**Review by hjorthjort — APPROVED** [review]
LGTM, but I'll let Andrei chime in if he wants to comment on the structure

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**